js/exporter/jspdf/export.js

```diff
@@ -125,24 +126,30 @@ export const Export = {
         }
     },
 
-    setColumnWidths: function(autoTableOptions, columns, keepColumnWidths) {
+    setColumnWidths: function(jsPDFDocument, autoTableOptions, columns, keepColumnWidths) {
         const columnStyles = autoTableOptions.columnStyles;
 
         for(let i = 0; i < columns.length; i++) {
             columnStyles[i] = columnStyles[i] || {};
 
-            const columnWidth = columns[i].gridColumn.width;
-            if(keepColumnWidths && (typeof columnWidth === 'number') && isFinite(columnWidth)) {
-                columnStyles[i].cellWidth = this.convertPixelsToPoints(columnWidth);
+            const isColumnWidthDefined = isDefined(columns[i].gridColumn.width);
+            if(keepColumnWidths && isColumnWidthDefined) {
+                columnStyles[i].cellWidth = this.convertPixelsToPdfUnits(jsPDFDocument, columns[i].width);
             } else {
                 columnStyles[i].cellWidth = 'auto';
             }
         }
     },
 
-    convertPixelsToPoints: function(value) {
+    convertPixelsToPdfUnits: function(pdfDoc, pxValue) {
+        const ptValue = this.convertPixelsToPoints(pxValue);
+        const scaleFactor = pdfDoc.internal.scaleFactor;
+        return ptValue / scaleFactor;
+    },
+
+    convertPixelsToPoints: function(pxValue) {
         const pointsPerInch = 72;
         const dotsPerInch = 96;
```

Comment:
magic :-(

Reply:
Maybe here you need to leave a comment with a link to the document how to convert units?
For example: https://www.answertabs.com/css-font-size-units-px-to-pt-in-em-rem-conversion/